networks/decoder_net.py

```diff
@@ -0,0 +1,130 @@
+"""This module contains the implementation of a decoder network, that applies 3 residual blocks
+to the input abstract position q. In the paper q is a (16, 4, 4) tensor that can be seen as a 4x4
+image with 16 channels, but here other sizes may be used.
+"""
+
+import torch
+from torch import nn
+
+
+class ResidualBlock(nn.Module):
+    """A residual block that up-samples the input image by a factor of 2.
+    """
+
+    def __init__(self, in_channels, n_filters=64, kernel_size=3, padding='same'):
+        """Instantiate the residual block, composed by a 2x up-sampling and two convolutional
+        layers.
+
+        Args:
+            in_channels (int): Number of input channels.
+            n_filters (int): Number of filters, and thus output channels.
+            kernel_size (int): Size of the convolutional kernels.
+            padding (Union[str, int]): Padding for the convolutional kernels. Can be an int,
+                or 'same' to automatically compute the padding to preserve the image size.
+        """
+        super().__init__()
+        self.channels = in_channels
+        self.n_filters = n_filters
+        if isinstance(padding, int):
+            use_padding = padding
+        elif padding == 'same':
+            use_padding = int(kernel_size / 2)
+        else:
+            raise ValueError('Padding argument not understood. Must be integer or "same".')
+        self.conv1 = nn.Conv2d(
+            in_channels=in_channels,
+            out_channels=n_filters,
+            kernel_size=kernel_size,
+            padding=use_padding,
+            bias=False
```

Comment:
No bias?

Reply:
I had copied the code and I forgot to change it. I think we should have biases